bolt/lib/Passes/PAuthGadgetScanner.cpp

```diff
@@ -787,19 +794,20 @@ shouldReportSigningOracle(const BinaryContext &BC, const MCInstReference &Inst,
                           const SrcState &S) {
   static const GadgetKind SigningOracleKind("signing oracle found");
 
-  MCPhysReg SignedReg = BC.MIB->getSignedReg(Inst);
-  if (SignedReg == BC.MIB->getNoRegister())
+  std::optional<MCPhysReg> SignedReg = BC.MIB->getSignedReg(Inst);
+  if (!SignedReg)
     return std::nullopt;
 
+  assert(*SignedReg != BC.MIB->getNoRegister());
```

Comment:
This assert looks a bit strange. Since we're using `std::optional<MCPhysReg>` everywhere now, is there any reason why `*SignedReg` could be `getNoRegister()` here more than in any other place where we call an API returning `std::optional<MCPhysReg>`?
If not, maybe it's better to remove this assert?

Reply:
Replacing `if (mayLoad(Inst)) {...}` with `if (Desc.mayLoad()) {...}` in the single-instruction overload of `getAuthCheckedReg` callback does not work: while `AArch64MCPlusBuilder::mayLoad(Inst)` returns true only for load instructions (but not all instructions are handled), the `MCInstrDesc::mayLoad()` method, in accordance with its name, returns true if the instruction *may* read anything from memory (such as for `AArch64::HINT`).

Added a FIXME in c73d6fdd679886c2177b06e1b4b806e90f59e7e4.